### PR TITLE
Iterative Execution of Walker/Downward

### LINKED ISSUE #202: Implement iterative execution of preprocessing and downward
So that we, say, run 10ms preprocessing then 10ms downward, then 100ms preprocessing then 100ms downward, etc
Until downward actually finds a result

### PR DESCRIPTION
**Pull request by kris701**
Also a massive refactor of a lot of components.
closes #202

### CODE REVIEW

File: src/Reformulators/WalkerReformulator.cpp [hunk 1]
```diff
@@ -2,21 +2,84 @@
...
+	auto ellapsed = Report->Stop(walkID);
+	if (Configs->GetItem<bool>("debugmode")) {
+		unsigned int totalIterations = paths.size();
+		unsigned int totalActionCount = walker->GetTotalActionsGenerated();
+		ConsoleHelper::PrintDebugInfo("[Walker] Total walk time:         " + to_string(ellapsed) + "ms", debugIndent);
+		double iterationsPrSecond = (totalIterations * 1000) / (ellapsed + 1);
+		ConsoleHelper::PrintDebugInfo("[Walker] Total walker iterations: " + to_string(totalIterations) + " [" + to_string(iterationsPrSecond) + "/s]", debugIndent);
+		double actionsPrSecond = (totalActionCount * 1000) / (ellapsed + 1);
+		ConsoleHelper::PrintDebugInfo("[Walker] Total actions Generated: " + to_string(totalActionCount) + " [" + to_string(actionsPrSecond) + "/s]", debugIndent);
+	}
```
Comment (jamadaha) [inline src/Reformulators/WalkerReformulator.cpp:22]: All these debug things might be better suited in functions.
It seems a tad odd, that a function with 3 lines of code performing actions is over 70 lines long.

File: settings.ini [hunk 2]
```diff
@@ -8,6 +8,9 @@ PATH:domain=Data/Classical tracks/Gripper/gripper_domain.pddl
 PATH:problem=Data/Classical tracks/Gripper/gripper_problem_hard.pddl
 PATH:downwardpath=../../P7Requirements/downward/fast-downward.py
 PATH:validatorpath=../../P7Requirements/downward/VAL/validate
+INT:startIncrement=1
+INT:incrementModifier=3
+INT:incrementLimit=120
```
Comment (jamadaha) [inline settings.ini:13]: These could really use an comments to explain what they are

File: src/CommonInterface/CommonInterface.cpp [hunk 3]
```diff
@@ -2,128 +2,241 @@
...
+		DownwardRunner runner = DownwardRunner();
+		runner.RunDownward(config, CommonInterface::TempDomainName, CommonInterface::TempProblemName, currentIncrementTimeLimit);
+		runRes = runner.ParseDownwardLog();
+		Report->Stop();
+		Report->Stop(iterationID);
+		if (runRes == DownwardRunner::FoundPlan) {
+			break;
+		}
+		currentIncrementTimeLimit *= config.GetItem<int>("incrementModifier");
+		counter++;
```
Comment (jamadaha) [inline src/CommonInterface/CommonInterface.cpp:106]: These lines are duplicate of 119 - 144?

File: src/CommonInterface/CommonInterface.cpp [hunk 4]
```diff
@@ -2,128 +2,241 @@
 
 using namespace std;
```
Comment (jamadaha) [inline src/CommonInterface/CommonInterface.cpp:3]: Might it be an idea to split all these functions into seperate classes, which all derive from a base class?
It seems that each has the same flow and duplicate lines, specifically print, report-start, ..., report-end. Thats over 30 lines? of code of the file which are duplicate

Comment (kris701) [reply src/CommonInterface/CommonInterface.cpp:3]: Im not sure i would say thats a good idea. One of the purposes of the common interface is that its easily modifiable and i think that making it into seperate classes would not only be a lot of work for little gain, but also make it more annoying to change anything in the pipeline in general.

**Review by jamadaha — APPROVED** [review]

**Pull request merged**